src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,59 @@
+package step2;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class StringAddCalculator {
+  private final Delimiters delimiters;
+
+  public StringAddCalculator() {
+    List<String> delimiters = new ArrayList<>();
+    delimiters.add(",");
+    delimiters.add(":");
+    this.delimiters = new Delimiters(delimiters);
+  }
+
+  public Integer calculate(String expression) {
+    if (expression == null || expression.trim().isEmpty()) {
+      return 0;
+    }
+
+    Map<String, String> derivedMap = deriveAdditionalDelimiter(expression);
+    this.delimiters.add(derivedMap.get("delimiter"));
+
+    /* #1
+    return this.delimit(derivedMap.get("expression")).stream()
+            .reduce(0, (a, b) -> b.sumInt(a), Integer::sum);
+
+    /* #2
+    return this.delimit(derivedMap.get("expression")).stream()
+            .reduce(new PositiveInteger(0), PositiveInteger::sum).getValue();
+
+     */
+
+    // #3
+    return this.delimit(derivedMap.get("expression")).stream()
+            .reduce(new PositiveNumber(0), PositiveNumber::add).getValue();
+  }
+
+  private List<PositiveNumber> delimit(String expression) {
+    String[] tokenized = expression.split(this.delimiters.getCombinedDelimiter());
+    return Arrays.stream(tokenized).map(PositiveNumber::new).collect(Collectors.toList());
+  }
+
+  private Map<String, String> deriveAdditionalDelimiter(String expression) {
+    if (expression.length() >= 4 && "//".equals(expression.substring(0, 2)) && "\n".equals(expression.substring(3, 4))) {
```

Comment:
```suggestion
    if (expression.length() >= 4 && "//".equals(expression.substring(0, 2)) && "\n".equals(expression.substring(3, 4))) {
```

어떠한 행동인지 한단어로 설명가능할까요? 메소드로 분리하면 어떠한 장점이 있을까요?

Reply:
각 조건을 메소드로 분리해서 이름을 지어주면, 굳이 주석이나 구두 설명이 없어도 이 조건문이 어떤 조건을 캐치하고자 하는지 명확히 알 수 있을 것 같습니다. 다르게 말하면 가독성이 좋아질 것 같습니다! 리뷰 감사드립니다 :)